node/silkworm/db/storage.cpp

```diff
@@ -20,126 +20,184 @@ limitations under the License.
 
 namespace silkworm::db {
 
-PruneMode read_prune_mode(mdbx::txn& txn) {
-    PruneMode ret{true};
-    auto src{db::open_cursor(txn, table::kDatabaseInfo)};
-
-    /*
-     * TODO Eventually optimize with removal of keys which don't prune at all
-     * For compatibility reasons with Erigon we assume a value == UINT64_MAX means no pruning
-     */
-
-    // History
-    auto data{src.find(mdbx::slice{kPruneModeHistoryKey}, /*throw_notfound=*/false)};
+//! \brief Retrieves the proper BlockAmount prune threshold for given key
+static BlockAmount read_block_amount_for_key(mdbx::cursor_managed& source, const char* key) {
```

Comment:
(Just nitpicking) It'd be cleaner to pass `source` as `mdbx::cursor&` rather than `mdbx::cursor_managed&` since this function doesn't care if the cursor as managed or not.

Reply:
Yes you're right
Doing